docs/glossary.rst

```diff
@@ -38,46 +24,57 @@ Astropy Glossary
        For example, ``distance : quantity-like ['length']``
 
    angle-like
-      :term:`quantity-like`, but interpreted by an angular
-      `~astropy.units.SpecificTypeQuantity`, like `~astropy.coordinates.Angle`
-      or `~astropy.coordinates.Longitude` or `~astropy.coordinates.Latitude`.
-      Note that the interpretation of units in strings depends on the class --
-      ``Quantity("180d")`` is 180 days, while ``Angle("180d")`` is 180 degrees
-      -- so make sure the string parses as intended for ``Angle``.
+      :term:`quantity-like` and a valid initializer for `~astropy.coordinates.Angle`.
+      The ``unit`` must be an angle equivalent to radians. A string input is interpreted
+      as an angle as described in the `~astropy.coordinates.Angle` documentation.
 
-   length-like
-      :term:`quantity-like`, but interpretable by
-      :class:`~astropy.coordinates.Distance`.
+   buffer-like
+      Object that implements `Python's buffer protocol
+      <https://docs.python.org/3/c-api/buffer.html#bufferobjects>`_.
+
+   coordinate-like
+      :class:`~astropy.coordinates.BaseCoordinateFrame` subclass instance, or a
+      :class:`~astropy.coordinates.SkyCoord` (or subclass) instance, or a valid
+      initializer as described in :ref:`coordinates-initialization-coord`.
+
+   file-like (readable)
+      :term:`python:file-like object` object that supports reading with a method ``read``.
+
+   file-like (writeable)
+      :term:`python:file-like object` object that supports writing with a method ``write``.
 
    frame-like
-      A :class:`~astropy.coordinates.BaseCoordinateFrame` subclass instance or a
-      string that can be converted to a Frame by
-      :class:`~astropy.coordinates.sky_coordinate_parsers._get_frame_class`.
+      :class:`~astropy.coordinates.BaseCoordinateFrame` subclass or subclass instance or
+      a valid Frame name (string).
 
-   coordinate-like
-      A Coordinate-type object such as a
-      :class:`~astropy.coordinates.BaseCoordinateFrame` subclass instance or a
-      :class:`~astropy.coordinates.SkyCoord` (or subclass) instance.
+   length-like
+      :term:`quantity-like` and a valid initializer for
+      :class:`~astropy.coordinates.Distance`. The ``unit`` must be a length equivalent
+      to meters.
 
-   table-like
-      An astropy :class:`~astropy.table.Table` or any object that can
-      initialize one. Anything marked as table-like will be processed through
-      a :class:`~astropy.table.Table`.
+   number
+      Any numeric type. e.g. `float` or `int` or ``numpy.number``.
```

Comment:
```suggestion
      Any scalar numeric type. e.g. `float` or `int` or ``numpy.number``.
```

Reply:
Just to better distinguish from the array types.

Reply:
Fixed.